src/main/java/org/jvnet/winp/Native.java

```diff
@@ -70,7 +71,20 @@
     private static final String DLL_TARGET = "winp.folder.preferred";
     private static final String UNPACK_DLL_TO_PARENT_DIR = "winp.unpack.dll.to.parent.dir";
 
-    private static String ctrlCExePath;
+    private static final LazyValue<String> ctrlCExePathLazyValue = new LazyValue<String>(new Callable<String>() {
```

Comment:
It might be useful from the common library use-cases, I'd guess. From the Jenkins standpoint I would prefer the error to happen during initialization of the class so that admins get notified early, but `static {` does not guarantee the invocation time also

Reply:
`ctrlCExePathLazyValue.getValue()` is called on the 85th line
Though, it might be not obvious. I'm thinking of simplifying this change: store the exception object directly in Native.

Reply:
Something like
```
Index: src/main/java/org/jvnet/winp/Native.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/java/org/jvnet/winp/Native.java	(revision c9537f464d40e8cd198e04086e0b03defbc37bc5)
+++ src/main/java/org/jvnet/winp/Native.java	(date 1548452527314)
@@ -82,6 +82,9 @@
      */
     @CheckReturnValue
     public static boolean sendCtrlC(int pid) throws WinpException {
+        if (loadFailure != null) {
+            throw new WinpException("Cannot send the CtrlC signal to the process: winp init failed", loadFailure);
+        }
         if (ctrlCExePath == null) {
             LOGGER.log(Level.WARNING, "Cannot send the CtrlC signal to the process. Cannot find the executable {0}.dll", CTRLCEXE_NAME);
             return false;
@@ -89,9 +92,16 @@
         return CtrlCSender.sendCtrlC(pid, ctrlCExePath);
     }
 
+    private static Throwable loadFailure;
+
     static {
-        File exeFile = load();
-        ctrlCExePath = (exeFile == null) ? null : exeFile.getPath();
+        try {
+            File exeFile = load();
+            ctrlCExePath = (exeFile == null) ? null : exeFile.getPath();
+        } catch (Throwable t) {
+            loadFailure = t;
+            LOGGER.log(Level.SEVERE, "Cannot init winp native", t);
+        }
     }
 
     private static String md5(URL res) {
```

If OK, I'll overwrite the change with force push.

Reply:
Missed line 85. Moving to Native might be preferable, but the current state also looks good to me

Reply:
Done (change simplified).

Reply:
@oleg-nenashev The change is simpler now. Please take a look.